File: VB/POCOBindableProperties/StandardDataBindingUserControl.vb
Imports DevExpress.MVVM
Imports DevExpress.MVVM.DataAnnotations
Imports DevExpress.MVVM.POCO
Imports DevExpress.Utils.MVVM
Imports DevExpress.XtraEditors
Imports System
Imports System.Windows.Forms

Namespace DxSamplePOCOBindableProperties
    Partial Public Class StandardDataBindingUserControl
        Inherits DevExpress.XtraEditors.XtraUserControl
        Public Sub New() 
            InitializeComponent()
            Dim mvvmContext As New MVVMContext()
            mvvmContext.ContainerControl = Me
        
            Dim editor As New TextEdit()
            editor.Dock = DockStyle.Top
            editor.Properties.NullValuePrompt = "Please, enter the Title here..."
            editor.Properties.NullValuePromptShowForEmptyValue = True
        
            Dim commandButton As New SimpleButton()
            commandButton.Dock = DockStyle.Top
            commandButton.Text = "Report the Title property value"
        
            commandButton.Parent = Me
            editor.Parent = Me

            ' Set type of POCO-ViewModel
            mvvmContext.ViewModelType = GetType(ViewModel)
            Dim viewModel As ViewModel = mvvmContext.GetViewModel(Of ViewModel)()
            ' Data binding for the Title property (via the DataBindings collection)
            editor.DataBindings.Add("EditValue", viewModel, "Title", True, DataSourceUpdateMode.OnPropertyChanged)
            ' UI binding for the Report command
            AddHandler commandButton.Click, Sub(s, e) XtraMessageBox.Show(viewModel.GetTitleAsHumanReadableString())
        End Sub
         Private Sub OnDisposing()
            Dim context = MVVMContext.FromControl(Me)
            If context IsNot Nothing Then
                context.Dispose()
            End If
        End Sub
        '' 
        '' POCO View Model provides out-of-the-box support of the INotifyPropertyChanged.
        '' 
        Public Class ViewModel
            ' Bindable property will be created from this property.
            Public Overridable Property Title() As String
            ' Just a method for readability
            Public Function GetTitleAsHumanReadableString() As String
                If Title Is Nothing Then
                    Return "(Null)"
                End If
                If Title.Length = 0 Then
                    Return "(Empty)"
                End If
                If String.IsNullOrWhiteSpace(Title) Then
                    Return "(WhiteSpace)"
                End If
                Return Title
            End Function
        End Class
    	End Class
End Namespace

File: CS/POCOBindableProperties/StandardDataBindingUserControl.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.POCO;
using DevExpress.Utils.MVVM;
using DevExpress.XtraEditors;
using System;
using System.Windows.Forms;

namespace DxSamplePOCOBindableProperties {
    public partial class StandardDataBindingUserControl : XtraUserControl {
        public StandardDataBindingUserControl() {
            InitializeComponent();
            #region SetUp
            MVVMContext mvvmContext = new MVVMContext();
            mvvmContext.ContainerControl = this;
        
            TextEdit editor = new TextEdit();
            editor.Dock = DockStyle.Top;
            editor.Properties.NullValuePrompt = "Please, enter the Title here...";
            editor.Properties.NullValuePromptShowForEmptyValue = true;
        
            SimpleButton commandButton = new SimpleButton();
            commandButton.Dock = DockStyle.Top;
            commandButton.Text = "Report the Title property value";
        
            commandButton.Parent = this;
            editor.Parent = this;
            #endregion SetUp

            #region #standardDataBinding
            // Set type of POCO-ViewModel
            mvvmContext.ViewModelType = typeof(ViewModel);
            ViewModel viewModel = mvvmContext.GetViewModel<ViewModel>();
            // Data binding for the Title property (via the DataBindings collection)
            editor.DataBindings.Add("EditValue", viewModel, "Title", true, DataSourceUpdateMode.OnPropertyChanged);
            // UI binding for the Report command
            commandButton.Click += (s, e) => XtraMessageBox.Show(viewModel.GetTitleAsHumanReadableString());
            #endregion #standardDataBinding
        }
        #region CleanUp
        void OnDisposing() {
            var context = MVVMContext.FromControl(this);
            if(context != null) context.Dispose();
        }
        #endregion CleanUp
        // 
        // POCO View Model provides out-of-the-box support of the INotifyPropertyChanged.
        // 
        public class ViewModel {
            // Bindable property will be created from this property.
            public virtual string Title { get; set; }
            // Just a method for readability
            public string GetTitleAsHumanReadableString() {
                if(Title == null)
                    return "(Null)";
                if(Title.Length == 0)
                    return "(Empty)";
                if(string.IsNullOrWhiteSpace(Title))
                    return "(WhiteSpace)";
                return Title;
            }
        }
    }
}